apps/expert/lib/expert/engine.ex

```diff
@@ -0,0 +1,224 @@
+defmodule Expert.Engine do
+  @moduledoc """
+  Utilities for managing Expert engine builds.
+
+  When Expert builds the engine for a project using Mix.install, it caches
+  the build in the user data directory. If engine dependencies change (e.g.,
+  in nightly builds), Mix.install may not know to rebuild, causing errors.
+
+  This module provides functions to inspect and clean these cached builds.
+  """
+
+  @doc """
+  Runs engine management commands based on parsed arguments.
+
+  Returns the exit code for the command. Clean operations will stop at the
+  first deletion error and return exit code 1.
+  """
+
+  @success_code 0
+  @error_code 1
+
+  @help_options ["-h", "--help"]
+
+  @spec run([String.t()]) :: non_neg_integer()
+  def run(args) do
+    {opts, subcommand, _invalid} =
+      OptionParser.parse(args,
+        strict: [force: :boolean],
+        aliases: [f: :force]
+      )
+
+    case subcommand do
+      ["ls"] -> list_engines()
+      ["ls", options] when options in @help_options -> print_ls_help()
```

Comment:
I believe the way to accomplish the "subcommand" cli we are wanting is to parse the options, get the subcommand, then parse the options again.

Also not we use parse_head instead of parse.

So for example

```elixir
{opts, rest, _invalid} =
      OptionParser.parse_head(args,
        strict: [help: :boolean],
        aliases: [h: :help]
      )

case rest do
  ["clean" | clean_opts] -> clean_engines(clean_opts) 
  # the rest
end

defp clean_engines(argv) do
  {opts, _rest, _invalid} =
      OptionParser.parse_head(argv,
        strict: [force: :boolean, help: :boolean],
        aliases: [f: :force, h: :help]
      )

  # do the rest

end
```